source/common/network/address_impl.cc

```diff
@@ -160,5 +160,49 @@ InstanceConstSharedPtr parseInternetAddress(const std::string& ip_addr) {
   return nullptr;
 }
 
+InstanceConstSharedPtr parseInternetAddressAndPort(const std::string& addr) {
```

Comment:
Looks good for the parsing code in address_impl.cc, but the test_common code isn't included in coverage AFAICT.